core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
I do understand that, it just seems to hinder composability. Any file system that for example transforms `Size` should know about the different kinds of size: `Eval[Size]`, `Size` and `Future[Size]`. Then, when you add a new effect, you need to update all file systems.

As far as `Future` is concerned we have a problem anyway because, if I'm not mistaken, fuse is blocking anyway. In that light I'm ok to try the direction you propose.

I however, as stated above, don't quite see how we can make filesystems that transforms a certain property when that property is potentially effectful, unless you provide all transformer filesystems with an `M[_]`. Moving the effects into the keys seems to make the 'transformer filesystems' like `filterfs`, `mapfs`, ... much harder to implement.

Anyway, I think all concerns are on the table. I'm happy to take the route you propose.